modules/tile-converter/src/i3s-converter/helpers/gltf-attributes.ts

```diff
@@ -49,35 +49,48 @@ export function prepareDataForAttributesConversion(tileContent: B3DMContent): B3
       };
     }) || [];
 
-  const prepearedNodes = nodes.map((node) => {
-    if (!node.mesh) {
-      return node;
-    }
-
-    return {
-      ...node,
-      images,
-      mesh: {
-        ...node.mesh,
-        primitives: node.mesh?.primitives.map((primitive) => ({
-          ...primitive,
-          indices: {value: primitive?.indices?.value},
-          attributes: getB3DMAttributesWithoutBufferView(primitive.attributes),
-          material: {
-            id: primitive?.material?.id
-          }
-        }))
-      }
-    };
-  });
+  prepareNodes(nodes, images);
 
   const cartographicOrigin = tileContent.cartographicOrigin;
   const cartesianModelMatrix = tileContent.cartesianModelMatrix;
 
   return {
     gltfMaterials,
-    nodes: prepearedNodes,
+    nodes,
     cartographicOrigin,
     cartesianModelMatrix
   };
 }
+
+/**
+ * Traverse all nodes to replace all sensible data with copy.
```

Comment:
Can we improve / expand this comment. What does "sensible data" mean? Why do we replace it with copies? Does this increase memory pressure?

Reply:
Updated comment. This function need to copy arrayBuffers to avoid data corruption in worker.